Philips.CodeAnalysis.Common/SingleDiagnosticAnalyzer.cs

```diff
@@ -15,14 +15,17 @@ namespace Philips.CodeAnalysis.Common
 	public abstract class SingleDiagnosticAnalyzer : DiagnosticAnalyzer
 	{
 		public DiagnosticId DiagnosticId { get; }
-
-		public DiagnosticDescriptor Rule { get; }
+		public string Id { get; }
+		protected Helper Helper { get; }
+		protected DiagnosticDescriptor Rule { get; }
 
 		protected SingleDiagnosticAnalyzer(DiagnosticId id, string title, string messageFormat, string description, string category,
-											DiagnosticSeverity severity = DiagnosticSeverity.Error, bool isEnabled = true)
+											Helper helper = null, DiagnosticSeverity severity = DiagnosticSeverity.Error, bool isEnabled = true)
```

Comment:
You could migrate `isEnabled` into a property and initialize it with `true`. If an implementations wants it otherwise, they can set the property in the constructor body. Might require the `Rule` property to be Lazy though.

Reply:
Consider the same for `severity`

Reply:
Yes, the immutable Rule must be created by the time the constructors complete. But I should be able to change the property to protected set, and then re-create the Rule in the leaf class' constructor for the special cases (severity/isEnabled not default). I'll play with that later today.